src/python/pants/help/scope_info_iterator.py

```diff
@@ -1,23 +1,23 @@
 # Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+from dataclasses import dataclass
+from typing import Any, Dict, Iterator, List, Set, Type
 
 from pants.option.global_options import GlobalOptionsRegistrar
 from pants.option.parser_hierarchy import enclosing_scope
 from pants.option.scope import GLOBAL_SCOPE, ScopeInfo
 from pants.subsystem.subsystem_client_mixin import SubsystemClientMixin
 
 
+@dataclass
 class ScopeInfoIterator:
     """Provides relevant ScopeInfo instances in a useful order."""
 
-    def __init__(self, scope_to_info):
-        """
-        :param dict scope_to_info: A map of scope name -> ScopeInfo instance.
-        """
-        self._scope_to_info = scope_to_info
+    scope_to_info: Dict[str, ScopeInfo]
+    v2_help: bool = False
 
-    def iterate(self, scopes):
+    def iterate(self, scopes: Set[str]) -> Iterator[ScopeInfo]:
```

Comment:
Do we actually care that it's a `Set`? If the only thing we ever do is `for s in scopes`, it's better to define this as `Iterable[str]` so that you can, for example, pass a generator expression.

Reply:
I think we actually do care that it's a set - we want to guarantee that the inputs to this function don't have duplicates, because if there are duplicates the code as written will end up printing multiple identical help sections one after another. If we really wanted to we could change the signature of `iterate` to accept a generic `Iterable[str]` and de-duplicate in the body of the method, but I think that requiring the input to be a `Set` specifically accomplishes the same goal with fewer code changes.

Reply:
Okay, that sounds reasonable.